### PR TITLE
Add support for XT8x

### PR DESCRIPTION
**Pull request by kopibeng**
## Description

Adds support for XT8x

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for XT8x

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kopibeng/mnk88/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = STM32F072
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/kopibeng/mnk88/rules.mk:7]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/kopibeng/mnk88/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+![MNK88](https://i.imgur.com/QlKuV3w.png)
+
+A QMK-powered, VIA-enabled 88-key (F13) TKL PCB with support for ANSI/ISO layouts, split Backspace, split Right Shift and 6.25U/7U bottom row.
+
+* Keyboard Maintainer: kopibeng
+* Hardware Supported: A TKL keyboard with STM32F072 MCU
+* Hardware Availability: https://monokei.co/
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/kopibeng/mnk88/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kopibeng/mnk88/rules.mk (hunk 1 above)
Comment (kopibeng) [reply keyboards/kopibeng/mnk88/rules.mk:7]: Updated, thanks.

File: keyboards/kopibeng/mnk88/readme.md (hunk 2 above)
Comment (kopibeng) [reply keyboards/kopibeng/mnk88/readme.md:11]: Updated, thanks.

**Comment by kopibeng** [comment]
It seems I have a duplicated keyboard (MNK88, https://github.com/qmk/qmk_firmware/pull/14435) PR in this ticket, how should I resolve this such that it is only related to XT8x PCB?

File: keyboards/kopibeng/xt8x/xt8x.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    bool res = led_update_user(led_state);
+
+    if(res) {
+        writePin(LED_CAPS_LOCK_PIN, led_state.caps_lock);
+        writePin(LED_SCROLL_LOCK_PIN, led_state.scroll_lock);
+    }
+    return res;
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
```
Comment (drashna) [inline keyboards/kopibeng/xt8x/xt8x.c:40]:
```suggestion
__attribute__((weak)) layer_state_t layer_state_set_user(layer_state_t state) {
```

Comment (kopibeng) [reply keyboards/kopibeng/xt8x/xt8x.c]: Updated, thanks.

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
> It seems I have a duplicated keyboard (MNK88, #14435) PR in this ticket, how should I resolve this such that it is only related to XT8x PCB?

You should be able to delete the board from the web editor, if you need.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**